File: src/web/Bc.Web/Controllers/CommentController.cs
using System;
using System.Threading.Tasks;
using Bc.Contracts.Internals.Endpoint.CommentTaking.Commands;
using Bc.Web.Models;
using Microsoft.AspNetCore.Mvc;
using NServiceBus;

namespace Bc.Web.Controllers
{
    [ApiController]
    [Route("[controller]")]
    public class CommentController : ControllerBase
    {
        private readonly IMessageSession messageSession;

        public CommentController(IMessageSession messageSession)
        {
            this.messageSession = messageSession;
        }

        [HttpPost]
        public async Task<ActionResult<Comment>> PostTodoItem(Comment comment)
        {
            // guess what for ;)
            if (!string.IsNullOrEmpty(comment.UserPhone))
            {
                return Ok();
            }

            var command = new TakeComment(
                    Guid.NewGuid(),
                    comment.UserName,
                    comment.UserEmail,
                    comment.UserWebsite,
                    comment.UserComment,
                    comment.ArticleFileName,
                    DateTime.UtcNow
                );

            await this.messageSession.Send(command).ConfigureAwait(false);

            return Ok();
        }
    }
}

File: src/web/Bc.Web/Controllers/CommentController.fs
namespace Bc.Web.Controllers

open System
open System.Threading.Tasks
open Bc.Contracts.Internals.Endpoint.CommentTaking.Commands
open Bc.Web.Dto
open Microsoft.AspNetCore.Mvc
open NServiceBus

[<ApiController>]
[<Route("[controller]")>]
type CommentController(messageSession: IMessageSession) =
    inherit ControllerBase()

    [<HttpPost>]
    member this.Post(comment: Comment) =

        // // guess what for ;)
        if not (String.IsNullOrEmpty(comment.UserPhone)) then
           base.Ok()
        else
            let command = TakeComment(
                            Guid.NewGuid(),
                            comment.UserName,
                            comment.UserEmail,
                            comment.UserWebsite,
                            comment.UserComment,
                            comment.ArticleFileName,
                            DateTime.UtcNow)

            messageSession.Send(command) |> Async.AwaitTask |> ignore
            
            base.Ok()

    [<HttpGet>]
    member this.Get() =
        "Ready."